agent/consul/leader.go

```diff
@@ -1084,7 +1084,7 @@ func parseCARoot(pemValue, provider, clusterID string) (*structs.CARoot, error)
 		ID:                  id,
 		Name:                fmt.Sprintf("%s CA Root Cert", strings.Title(provider)),
 		SerialNumber:        rootCert.SerialNumber.Uint64(),
-		SigningKeyID:        connect.HexString(rootCert.AuthorityKeyId),
+		SigningKeyID:        connect.HexString(rootCert.SigningKeyId),
```

Comment:
@tradel did this compile for you?

`rootCert` is a `*x509.Certificate` which has no field called `SigningKeyId`. https://golang.org/pkg/crypto/x509/#Certificate

The CI build doesn't compile this.

I this the confusion here comes from the fact that in the client we are handling responses of type `structs.CARoot` which has this `SigningKeyID` field. But that should correspond to the `AuthorityKeyID` of the _leaf_ cert.

I.e. each leaf should have an `AuthorityKeyID` which is the ID of the certificate that signed it.

So I _think_ the bug here is that for a self-signed root, the signing authority (`AuthorityKeyID`) and the actual root cert's ID (`SubjectKeyID`) are the same. But for an intermediate, they are different and we really want the `SubjectKeyID` here.

To confuse matters. We are using the word "Root" here to mean "a Connect trust root" and something that can sign certs. It might not actually be an x509 root and so may be an intermediate cert. In that case we should be using the ID of the signing cert (the one in the `pemValue` here) as the `SigningKeyID` not the ID of they key that signed the "root".

tl;dr:
```suggestion
		SigningKeyID:        connect.HexString(rootCert.SubjectKeyID),
```

But tests would be good too.